cedar-lean-cli/src/util.rs

```diff
@@ -193,39 +191,31 @@ impl OpenRequestEnv {
                 Some(req_act_id) if (*req_act_id).id() != action_id.id() => continue,
                 _ => (),
             };
-            match schema.principals_for_action(&action_id) {
-                Some(principal_types) => {
-                    for principal_type in principal_types {
-                        match &self.principal_type {
-                            Some(req_principal_type) if req_principal_type != principal_type => {
-                                continue
-                            }
-                            _ => (),
-                        };
-                        match schema.resources_for_action(&action_id) {
-                            Some(resource_types) => {
-                                for resource_type in resource_types {
-                                    match &self.resource_type {
-                                        Some(req_resource_type)
-                                            if req_resource_type != resource_type =>
-                                        {
-                                            continue
-                                        }
-                                        _ => (),
-                                    };
-                                    let req_env = RequestEnv::new(
-                                        principal_type.clone(),
-                                        action_id.clone(),
-                                        resource_type.clone(),
-                                    );
-                                    req_envs.push(req_env)
+            if let Some(principal_types) = schema.principals_for_action(&action_id) {
+                for principal_type in principal_types {
+                    match &self.principal_type {
+                        Some(req_principal_type) if req_principal_type != principal_type => {
+                            continue
+                        }
+                        _ => (),
+                    };
```

Comment:
```suggestion
if let Some(req_principal_type) = &self.principal_type && req_principal_type != principal_type {
    continue;
}
```

Reply:
Also, should we check and extract the value in `&self.principal_type` before going into this loop? Then you wouldn't need to have an `if let Some(...)` check in each iteration.